cloudevents-sdk-reqwest/src/client_response.rs

```diff
@@ -10,6 +11,24 @@ use reqwest::header::{HeaderMap, HeaderName};
 use reqwest::Response;
 use std::convert::TryFrom;
 
+/// Extention Trait for [`Response`]
+#[async_trait]
```

Comment:
I'm trying to wonder if we can skip using the `async_trait` crate using a generic. I'll try locally and i'll let you know

Reply:
Alright! But from what I know, you can't define `async` functions for a Trait function without `#[async_trait]`

Reply:
yes but `async` is a syntactic sugar for <T: Future<...>> fn() -> T, so in theory it should be doable